android/app/buildConfigs.gradle

```diff
@@ -1,58 +1,117 @@
-// See https://github.com/luggit/react-native-config/blob/master/android/dotenv.gradle for reference
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath group: 'org.apache.commons', name: 'commons-lang3', version: '3.11'
+    }
+}
 
-def createConfig () {
-    def buildConfig = [:]
+import groovy.json.JsonSlurper
 
-    if (!System.env["BUILD_CONFIG_NAME"]) {
-        println("*****************++++++++*********")
-        println("*** Missing BUILD_CONFIG_NAME ****")
-        println("************************++++++++**")
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.stream.Collectors
+import org.apache.commons.lang3.StringUtils
 
-        // Set to empty array to prevent failing for other tasks
-        project.ext.set("buildConfig", buildConfig)
+def createConfig() {
+    def buildConfigName
 
-        return
+    if (!System.env["BUILD_CONFIG_NAME"]) {
+        buildConfigName = 'integreat'
+    } else {
+        logger.warn("WARNING: No build config specified. Using the build config 'integreat'!")
+        buildConfigName = System.env['BUILD_CONFIG_NAME']
     }
 
-    def buildConfigName = System.env['BUILD_CONFIG_NAME']
-    println("Using build config $buildConfigName")
+    logger.quiet("Using build config $buildConfigName")
 
-    def command = "yarn babel-node tools/create-xcconfigs config $buildConfigName --directory android"
+    def command = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
     def proc = command.execute()
     proc.waitFor()
 
     if (proc.exitValue() != 0) {
-        println("Failed to create xcconfig file: $proc.err.text")
+        logger.error("Failed to get build config: $proc.err.text")
         return
     }
-    println("Created xcconfg file: $proc.in.text")
+    def json = proc.in.text
+    logger.quiet("Build Config as JSON: $json")
 
-    def buildConfigProperties = new Properties()
-    def buildConfigInputStream = new FileInputStream("$project.rootDir/buildConfig.tmp.xcconfig")
-    buildConfigProperties.load(buildConfigInputStream)
-
-    // Java properties use the same syntax as xcconfig files
-    // https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html
-    buildConfigProperties.each {
-        buildConfig.put(it.getKey(), it.getValue())
-    }
+    def buildConfig = new JsonSlurper().parseText(json)
 
-    buildConfigInputStream.close()
-
-    println("Build config: $buildConfig")
     project.ext.set("buildConfig", buildConfig)
 }
 
 createConfig()
 
+def joinKeys(List<String> keys) {
+    return String.join("_", keys.stream().map({ key -> decamelize(key) }).collect(Collectors.toList())).toUpperCase()
+}
+
+def decamelize(String key) {
+    return String.join("_", StringUtils.splitByCharacterTypeCamelCase(key))
+}
+
+def flattenBuildConfig(keys, buildConfig, resValue) {
```

Comment:
maybe you could use ´´´JsonGenerator.Options()´´´ here, as mentioned [here](https://stackoverflow.com/a/59103551). This would at least remove the string conversions here.

Reply:
hmm. maybe it would still be better to do that similar to the XCconfig on the js side. It would be more consistent and I think with lodash and js string parsing this method could be simplified a lot.

Reply:
I went with this way because it offers type safety. If you parse the XConfig file then everything is a string. This causes quite ugly checks which are not necessary as we have the type information.

Reply:
@Taggotty not sure what you mean here. ´JsonGenerator` does not parse jsons afaiks

Reply:
If I understood @Taggotty correctly, the camelize and upper case functionality should and could be done with js (script) which can then be used here and in the `create-xcconfig` script. I'd favor that to.